raster-foundry/raster-foundry
# Project layer UI cleanup

## Overview

This PR cleans up the project-layer UI in a few ways:

Styling the project layer analysis creation page
![image](https://user-images.githubusercontent.com/2442245/55012702-30a4d480-4fbe-11e9-8b7e-6898c5dba15e.png)

Using layer start and end dates where available otherwise not showing a date on layer items
![image](https://user-images.githubusercontent.com/2442245/55012958-98f3b600-4fbe-11e9-8591-a917a09c95b1.png)

Fixing style of the list action header to be more visibly separated from a list of things when scrolling and unifies button size usage within:
![image](https://user-images.githubusercontent.com/2442245/55013051-bfb1ec80-4fbe-11e9-99fd-b75df7eabf5c.png)

Allows clicking on project-layer names to go to the layer:
![image](https://user-images.githubusercontent.com/2442245/55020278-e840e300-4fcc-11e9-85fe-e0c79f76784e.png)

Updates the layer AOI UI to require less clicks:
![image](https://user-images.githubusercontent.com/2442245/55021287-de1fe400-4fce-11e9-8caf-5c533594b934.png)

### Checklist


### Notes

Added a new button class `btn-lightest` which uses existing color variables, so it is theme-safe.


## Testing Instructions

Testing this is a little hard, because it is mostly about trying to find style regressions.

 * Click around the v2 front-end and make sure there are no noticeably broken styles
 * Try clicking on the name of project layers to go to the layer
 * Create an analysis for a layer and make sure everything still works
 * Create/Edit layer AOIs and verify that everything is still usable
  - draw some things but don't confirm and then hit "< back" and make sure layers are cleared
  - edit an aoi shape and make sure only editing related buttons are visible
  - make sure you can't see any buttons other than "Draw AOI" when no AOI exists
  - make sure that confirming will save the AOI
  - try "save as shape & confirm" and make sure it works as expected

Closes #4743
